lib/matplotlib/legend.py

```diff
@@ -1260,6 +1260,8 @@ def _parse_legend_args(axs, *args, handles=None, labels=None, **kwargs):
     elif len(args) >= 2:
         handles, labels = args[:2]
         extra_args = args[2:]
+    elif len(handles)!= len(labels):
+        raise ValueError("fewer handles than labels or vice versa")
```

Comment:
It would be helpful to provide a better error message. Maybe include the length of handles and labels to make it easier for the developer to understand what they did wrong.
```suggestion
        raise ValueError(f"Mismatched number of handles and labels. len(handles)={len(handles)} and len(labels)={len(lables)}.")
```

Reply:
Sorry didn't mean to set myself as a reviewer. I just wanted to give my comments to @tusharkulkarni008 since it is their first PR ever.

Reply:
@AqeelAT the developer guide encourages you to review 🙂
https://matplotlib.org/stable/devel/triage.html#working-on-prs-to-help-review

Reply:
> Sorry didn't mean to set myself as a reviewer. I just wanted to give my comments to @tusharkulkarni008 since it is their first PR ever.

Yes @AqeelAT this is my first PR ever and earlier in this conversation other senior member pointed out some important in points in very detailed explanation. Since then I am observing PRs other people make to gain some footing to make more meaningful PRs. 
Certainly your error message is more consistent with raised issue